### PR TITLE
Removing Singularity and related linting

### PR DESCRIPTION
**Pull request by matq007**
Removing Singularity file together with linting functions related to it. I've also updated the doc. Should be everything hopefully. Implementation of issue #252.

### CODE REVIEW

**Comment by matq007** [comment]
I knew I should have tested before pushing 😿

File: docs/lint_errors.md [hunk 1]
```diff
@@ -209,50 +201,11 @@ These lines must be an exact copy of the above example.
...
-* `Bootstrap:docker`
-* `VERSION [pipeline version]`
-* `environment.yml /`
-* `/opt/conda/bin/conda env update -n root -f /environment.yml`
-* `/opt/conda/bin/conda clean -a`
-
-Additional lines and different metadata can be added without causing the test to fail.
-
-## Error #11 - Template TODO statement found ## {#11}
+## Error #10 - Template TODO statement found ## {#11}
```
[the remark this answers is not in the document]
Comment (ewels) [reply docs/lint_errors.md]: Also you need to update the error ID in the line that throws the error / passes for this test. eg:

```python
self.failed.append((11, "template todo something was wrong or something: {}".format(missing)))
```

File: docs/lint_errors.md [hunk 2]
```diff
@@ -6,46 +6,44 @@ This page contains detailed descriptions of the tests done by the [nf-core/tools
 nf-core pipelines should adhere to a common file structure for consistency. The lint test looks for the following required files:
 
 * `nextflow.config`
-  * The main nextflow config file
+    * The main nextflow config file
```
Comment (ewels) [inline docs/lint_errors.md]: You reverted all of my markdown fixes! 😱 I think the markdown linter will fail with this as it is..

Comment (matq007) [reply docs/lint_errors.md]: My editor ignored my settings 😱. Fixing

File: docs/lint_errors.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (matq007) [reply docs/lint_errors.md]: Isn't that defined here? https://github.com/nf-core/tools/blob/9f11213f2d60a6fdce86600ea086a4dc882daa8a/nf_core/lint.py#L766

**Comment by ewels** [comment]
This pull request **introduces 2 alerts** when merging 9f11213f2d60a6fdce86600ea086a4dc882daa8a into a1a7408ce7e7b4d502d29107bdea2cc406bb34f5 - [view on LGTM.com](https://lgtm.com/projects/g/nf-core/tools/rev/pr-bfd5aa87e43693576746033efc864c8010482015)

**new alerts:**

* 2 for Unused local variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: docs/lint_errors.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ewels) [reply docs/lint_errors.md]: Yes - but it's number 11. You need to change it to number 10.

[the remark this answers is not in the document]
Comment (matq007) [reply docs/lint_errors.md]: Oh, I see now Facepalm

**Comment by matq007** [comment]
I had to adjust some values in the tests.
- I've removed 3 tests so that means `MAX_PASS_CHECKS` should be only `58`
- I've removed Singularity file meaning that from passed tests we have to subtract 5 and then 1

**Comment by ewels** [comment]
Thanks! Yes the `MAX_PASS_CHECKS` thing is a bit of a hassle, currently just needs a manual update whenever you add or remove tests as you found. Not sure that there's a reliable better way to do it.

File: docs/lint_errors.md [hunk 3]
```diff
@@ -67,11 +65,11 @@ The following variables fail the test if missing:
   * The version of this pipeline. This should correspond to a [GitHub release](https://help.github.com/articles/creating-releases/).
 * `manifest.nextflowVersion`
   * The minimum version of Nextflow required to run the pipeline.
-  * Should `>=` a version number, eg. `manifest.nextflowVersion = '>=0.31.0'` (check the [Nexftlow documentation](https://www.nextflow.io/docs/latest/config.html#scope-manifest) for more.)
-  * This should correspond to the `NXF_VER` version tested by Travis.
+    * Should `>=` a version number, eg. `manifest.nextflowVersion = '>=0.31.0'` (check the [Nexftlow documentation](https://www.nextflow.io/docs/latest/config.html#scope-manifest) for more.)
```
Comment (ewels) [inline docs/lint_errors.md]: Still lots of whitespace changes in this diff...?

File: nf_core/lint.py [hunk 4]
```diff
@@ -757,39 +737,10 @@ def check_conda_dockerfile(self):
...
-        ]
-
-        difference = set(expected_strings) - set(self.singularityfile)
-        if not difference:
-            self.passed.append((10, "Found all expected strings in Singularity file"))
+            self.passed.append((9, "Found all expected strings in Dockerfile file"))
         else:
             for missing in difference:
-                self.failed.append((10, "Could not find Singularity file string: {}".format(missing)))
+                self.failed.append((9, "Could not find Dockerfile file string: {}".format(missing)))
```
Comment (ewels) [inline nf_core/lint.py]: I'm pretty sure that the Dockerfile test is still number 10... right?

Comment (matq007) [reply nf_core/lint.py]: No, it's 9. https://github.com/matq007/tools/blob/master/docs/lint_errors.md

Comment (ewels) [reply nf_core/lint.py]: Aha, nice! So this was an error in the previous code? Good spot 👍

File: docs/lint_errors.md [hunk 5]
```diff
@@ -83,9 +81,9 @@ The following variables throw warnings if missing:
   * The filename of the main pipeline script (recommended to be `main.nf`)
 * `timeline.file`, `trace.file`, `report.file`, `dag.file`
   * Default filenames for the timeline, trace and report
-  * Should be set to a results folder, eg: `${params.outdir}/pipeline_info/trace.[workflowname].txt"``
-  * The DAG file path should end with `.svg`
-    * If Graphviz is not installed, Nextflow will generate a `.dot` file instead
+    * Should be set to a results folder, eg: `${params.outdir}/pipeline_info/trace.[workflowname].txt"``
```
Comment (ewels) [inline docs/lint_errors.md]:
```suggestion
  * Should be set to a results folder, eg: `${params.outdir}/pipeline_info/trace.[workflowname].txt"`
```

Still some whitespace left which breaks indentation. Also spotted this broken inline code block which was there previously.

**Pull request merged**